src/wazuh_db/wdb_agent.c

```diff
@@ -430,14 +431,24 @@ int wdb_find_agent(const char *name, const char *ip) {
     if (wdb_open_global() < 0)
         return -1;
 
-    if (wdb_prepare(wdb_global, SQL_FIND_AGENT, -1, &stmt, NULL)) {
-        mdebug1("SQLite: %s", sqlite3_errmsg(wdb_global));
-        wdb_close_global();
-        return -1;
+    if (ip) {
```

Comment:
if we change ` wdb_find_agent(name, NULL))` for ` wdb_find_agent(name, addr))` the distintion does not seem necessary anymore